apps/designer/app/designer/features/style-panel/sections/spacing/input-popover.tsx

```diff
@@ -0,0 +1,118 @@
+import { useState } from "react";
+import {
+  keyframes,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverPortal,
+  styled,
+} from "@webstudio-is/design-system";
+import {
+  CssValueInput,
+  type IntermediateStyleValue,
+} from "../../shared/css-value-input";
+import type { StyleProperty, StyleValue } from "@webstudio-is/css-data";
+import type { StyleChangeEvent } from "./types";
+
+const slideUpAndFade = keyframes({
+  "0%": { opacity: 0, transform: "scale(0)" },
+  "100%": { opacity: 1, transform: "scale(1)" },
+});
+
+const Input = ({
+  value,
+  property,
+  onChange,
+  onClosePopover,
+}: {
+  property: StyleProperty;
+  value: StyleValue;
+  onChange: (event: StyleChangeEvent) => void;
+  onClosePopover: () => void;
+}) => {
+  const [intermediateValue, setIntermediateValue] = useState<
+    StyleValue | IntermediateStyleValue
+  >();
+
+  return (
+    <CssValueInput
+      property={property}
+      value={value}
+      intermediateValue={intermediateValue}
+      onChange={(styleValue) => {
+        setIntermediateValue(styleValue);
+        if (styleValue.type !== "intermediate") {
+          onChange({ property, value: styleValue, isEphemeral: true });
+        }
+      }}
+      onHighlight={(styleValue) => {
+        onChange({ property, value: styleValue, isEphemeral: true });
+      }}
+      onChangeComplete={(styleValue, reason) => {
+        setIntermediateValue(undefined);
+        onChange({ property, value: styleValue, isEphemeral: false });
+
+        // @todo: handle Esc
+        if (reason === "blur" || reason === "enter") {
+          onClosePopover();
+        }
+      }}
+    />
+  );
+};
+
+// trigger is used only for positioning
+const Trigger = styled("div", { position: "absolute", width: 0, height: 0 });
+
+const PopoverContentStyled = styled(PopoverContent, {
+  minWidth: 0,
+  minHeight: 0,
+  width: "$spacing$20",
+  border: "1px solid $colors$slate8",
+  borderRadius: "$borderRadius$7",
+  background: "$colors$gray2",
+  padding: "$spacing$5",
+  boxShadow: "0px 2px 7px rgba(0, 0, 0, 0.1), 0px 5px 17px rgba(0, 0, 0, 0.15)",
+  animationDuration: "150ms",
```

Comment:
this didn't help, the initial start is a bit too slow before it shows up, probably need to start with scale not 0 but something like 20-40, I will experiment locally a bit

Reply:
Yeah, please experiment yourself, looks all the same to me 😅